lib/apiservers/engine/backends/container.go

```diff
@@ -411,7 +419,8 @@ func (c *Container) ContainerRm(name string, config *types.ContainerRmConfig) er
 	if err != nil {
 		switch err := err.(type) {
 		case *containers.ContainerRemoveNotFound:
-			return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
+			cache.ContainerCache().DeleteContainer(name)
+			return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", paramName))
```

Comment:
Doesn't need to be for this PR, but we should really move "No such container" and similar strings that are repeated in many locations into the message catalogue.